### DISCUSSION TITLE
x-mask dropping digits when applying autocomplete

### DISCUSSION THREAD

**Discussion by robertmarney** [alpinejs/alpine#4252] [7. Bugs (archived)]
`'x-mask' => '+1 (999) 999-9999'`

User provided input of 1234567890 is being applied as +1 (234) 567-890, where we would expect +1 (123) 456-7890

This is replicated here without any external dependencies / wrappers.

https://codepen.io/Robert-Marney/pen/RwmVqWE

You can see that if the mask does not start with a number it works fine (e.g. +z (999) 999-9999)

For clarity any element of the mask that is not `9` `a` or `*` should be treated as the mask.

**Comment by ekwoka** [comment]
I would expect the first.

Since that is 10 numbers, not 11.

So the first 1 is consumed in the mask itself.

Luckily, there are no valid area codes that begin with 1, so this isn't a real issue of concern.

> For clarity any element of the mask that is not 9 a or * should be treated as the mask.

it is.

so the mask strips itself out, which gives you `234567890` since the `1` matches to the mask. Then it reconstructs the string with the mask to `+1 (234) 567-890`
